File: TOP150Review/Minimum Size Subarray Sum.py

# sliding window , easy to keep track the window content 
class Solution:

    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        
        left = 0 
        right = 0 
        window = 0 
        sol = float("inf") 

        # use open interval [left , right) 
        while right < len(nums): 

            window += nums[right]
            right += 1 

            
            while left < right and window >= target : 
                sol = min(  right-left   , sol  )
                window -= nums[left]
                left += 1 
            
        return sol if sol != float("inf") else 0 

File: RUSH/Minimum Size Subarray Sum.js
/**
 * @param {number} target
 * @param {number[]} nums
 * @return {number}
 */

// sliding-window approach 
var minSubArrayLen = function(target, nums) {
    
    let window = 0 
    let left = 0 
    let right = 0 
    let solution = Number.MAX_VALUE ;

    while (right < nums.length){

        window += nums[right] ; 
        right += 1 

        while ( window >= target && left < right ){

            solution = Math.min( solution , right-left )
            window -= nums[left]
            left += 1 

        }

    }

    return solution === Number.MAX_VALUE?  0 : solution ; 

};